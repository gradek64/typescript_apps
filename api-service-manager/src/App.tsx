import React, { useState } from 'react';
import './App.css';
import { APIResult, Endpoints } from './constants';
import generalAPIService from '../src/services/apiService';
const App: React.FC = () => {
  const [click, setClick] = useState(false);
  // wrapping fetch in a handleFetch to prevent it firing on page load
  //const handleFetch = (() => {

  const setMe = (event: React.MouseEvent): void => {
    event.preventDefault();
    setClick(!click);
  };

  const findCookie = (name: string) => {
    const cookies = document.cookie;
    if (cookies) {
      const cookie = document.cookie
        .split('; ')
        .find((row) => row.startsWith(name));
      if (cookie) {
        return cookie.split('=')[1];
      }
      return `Cookie with name ${name} doesnt exists`;
    }
    return `Cookie with name ${name} doesnt exists`;
  };

  // WEATHER API

  generalAPIService
    //this service accepts path,query, customHeader
    .get(
      '/api/location/search/',
      { params: { query: 'london' } },
      { customRequestHeaders: { greg: 'greg', bushra: 'bushra' } }
    )
    .then((data) => {
      console.log(data);
    })
    .catch((e) => {
      console.log(e);
    });

  //USERS and POSTS API
  //get users
  generalAPIService
    .get('/api/users/')
    .then((data) => {
      console.log(data);
    })
    .catch((e) => {
      console.log(e);
    });

  //sign in user
  generalAPIService
    .post('/api/users/signin', {
      body: {
        username: 'test',
        password: 'test',
      },
    })
    .then((data) => {
      //comes back with token that needs
      // sign in cookie
      document.cookie = `bearer=${data.token}`;
    })
    .catch((e) => {
      console.log(e);
    });

  //call this one with dealy since you need token
  setTimeout(async () => {
    const authCookie = findCookie('bearer');
    const authorized = {
      customRequestHeaders: { Authorization: `Bearer ${authCookie}` },
    };

    //create a post needs token for this route
    //set in headers
    const createPost = await generalAPIService.post(
      '/api/posts/create',
      {
        body: {
          title: 'new Post1',
          text: 'new Post Text1',
        },
      },
      authorized
    );

    console.log(createPost);
    //update the post needs token for this route and post id from previous request
    const postID = createPost._id;
    const updatePost = await generalAPIService.put(
      `/api/posts/${postID}`,
      {
        body: {
          title: 'new Post1 update',
          text: 'this is new Post text1 update',
        },
      },
      authorized
    );
    console.log(updatePost);
  }, 2000);

  return (
    <div className="App">
      <header className="App-header">
        <p onClick={setMe}>this is click event</p>
      </header>
    </div>
  );
};

export default App;
