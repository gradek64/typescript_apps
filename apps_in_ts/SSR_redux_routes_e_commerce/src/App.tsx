import React from 'react';
import { Provider } from 'react-redux';
import { Router } from 'react-router-dom'
import { createBrowserHistory } from "history";
// custom
import DisplayProductsList from './components/productsComposition/DisplayProductsList/DisplayProductsList';
import BasketIconStore from './components/functional/BasketIcon/BasketIcon';
import Modal from './common/Modal/Modal'
import Basket from './components/logic/Basket/Basket'
import store from './redux/store';
import customEvent from './customEvents/dispatchEvent'
import { getRoutes } from './shared/getRoutes'
import './app.scss'

const history = createBrowserHistory();


function App() {
  return (
    <Provider store={store}>
      <Router history={history}>{getRoutes(store)}
        {/* <header>
          <h2>Top Seller</h2>
        </header>
        <nav>
          <BasketIconStore eventHandler={() => customEvent.emit('OPEN_BASKET')} />
        </nav>
        <Modal eventOn="OPEN_BASKET">
          <Basket />
        </Modal>
        <DisplayProductsList /> */}
      </Router>
    </Provider>
  );
}

export default App;
