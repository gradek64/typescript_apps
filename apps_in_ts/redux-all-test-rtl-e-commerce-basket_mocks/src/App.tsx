import React from 'react';
import { Provider } from 'react-redux';
// custom
import DisplayProductsList from './components/productsComposition/DisplayProductsList/DisplayProductsList';
import BasketIconStore from './components/functional/BasketIcon/BasketIcon';
import Modal from './common/Modal/Modal'
import Basket from './components/logic/Basket/Basket'
import store from './redux/store';
import customEvent from './customEvents/dispatchEvent'
import './app.scss'


function App() {
  return (
    <Provider store={store}>
      <header>
        <h2>Top Seller</h2>
      </header>
      <nav>
        <BasketIconStore eventHandler={() => customEvent.emit('OPEN_BASKET')} />
      </nav>
      <Modal eventOn="OPEN_BASKET">
        <Basket />
      </Modal>
      <DisplayProductsList />

    </Provider>
  );
}

export default App;
