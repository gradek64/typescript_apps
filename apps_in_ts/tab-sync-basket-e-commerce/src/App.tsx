import React from 'react';
import { Provider } from 'react-redux';
// custom
import DisplayProductsList from './components/productsComposition/DisplayProductsList/DisplayProductsList';
import BasketIconStore from './components/functional/BasketIcon/BasketIcon';
import Modal from './common/Modal/Modal'
import Basket from './components/logic/Basket/Basket'
import store from './redux/store';
import customEvent from './customEvents/dispatchEvent'
import SelectDestinationCountry from './components/functional/SelectCountryDestionationContainer'
import './app.scss'

//multiTabSynchronisation
import { setUpMultiTabSyncListeners } from './redux/multiTabSync/multiTabSyncActions'

setUpMultiTabSyncListeners(store.dispatch, store.getState)

function App() {
  return (
    <Provider store={store}>
      <header>
        <h2>Top Seller</h2>
      </header>
      <nav>
        <SelectDestinationCountry />
        <BasketIconStore eventHandler={() => customEvent.emit('OPEN_BASKET')} />
      </nav>
      <Modal eventOn="OPEN_BASKET">
        <Basket />
      </Modal>
      <DisplayProductsList />

    </Provider>
  );
}

export default App;
