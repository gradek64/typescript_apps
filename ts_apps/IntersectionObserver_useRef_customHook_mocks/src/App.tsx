import React, { useState } from 'react';
import Grid from './Components/GridList';
import LinearProgress from '@material-ui/core/LinearProgress';
//custom
import useCallAPI from './customHooks/callAPI';
const classification = 'Prints';
const apikey = 'c28e4be0-4c0e-11ea-90d6-25d9a9fe80fc';
const totalpageviews = '1';
const size = '10';
const query = `${classification}&q=totalpageviews:${totalpageviews}&apikey=${apikey}&size=${size}`;
function App() {
  const [pageNumber, setPageNumber] = useState(1);
  console.log('pageNumber......', pageNumber);
  const { loading, error, dataSet } = useCallAPI(query, String(pageNumber));

  const updatePage = () => {
    setPageNumber((prevPageNumber) => prevPageNumber + 1);
  };
  return (
    <>
      {dataSet.length > 1 && <Grid dataSet={dataSet} updatePage={updatePage} />}
      {loading && 'loading...'}
      {error && 'error'}
    </>
  );
}

export default App;
