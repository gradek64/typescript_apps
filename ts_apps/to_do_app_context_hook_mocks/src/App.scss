@import './variables';
.container {
  padding: 20px;
  background-color: #3b2b2e;
  width: 50%;
  margin: 10% auto;
  min-width: 490px;
  border-radius: 10px;
}

@media only screen and (max-width: $mobile) {
  .container {
    width: 90%;
    min-width: 200px;
  }
}
