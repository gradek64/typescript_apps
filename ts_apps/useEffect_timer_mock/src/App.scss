body {
  margin: 0;
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen',
    'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue',
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New',
    monospace;
}

//avatar githubs

.container {
  display: flex;
  width: 100%;
  .images-container {
    padding: 10px;
    max-width: 80px;
    img {
      width: 100%;
      height: auto;

      border-radius: 50%;
    }
  }
  .text-container {
    h4,
    p {
      margin: 0;
      padding: 0;
    }
    padding: 10px;
  }
}
