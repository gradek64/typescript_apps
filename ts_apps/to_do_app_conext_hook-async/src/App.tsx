import React from 'react';
import './App.scss';

//custom
import TaskProvider from './components/Context/TaskProvider';
import AddToDo from './components/toDoListComp/AddToDo/AddToDo';
import TaskList from './components/toDoListComp/ToDoList/TaskList';

function App() {
  return (
    <TaskProvider>
      <div className="container">
        <div>
          <AddToDo />
        </div>
        <div>
          <TaskList />
        </div>
      </div>
    </TaskProvider>
  );
}

export default App;
