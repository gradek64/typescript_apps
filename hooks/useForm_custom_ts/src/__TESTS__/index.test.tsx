import React from 'react';
import {
  render,
  cleanup,
  fireEvent,
  queryByAttribute,
} from '@testing-library/react';
//import { shallow } from 'enzyme';
import '@testing-library/jest-dom/extend-expect';
import App from '../app';
import customEvent from '../customEvents/dispatchEvent';

describe('should render correclty with initial state', () => {
  afterEach(cleanup);
  it('should take a snapshot', () => {
    const { asFragment } = render(<App />);
    const component = asFragment();
    expect(component).toMatchSnapshot();
  });
});

describe("onSumbit all 'button' form", () => {
  it('should call handleSubmit event handler', () => {
    //use container cheaky way to have access to standard DOM selectors
    const { container } = render(<App />);
    //U need this hellper to access dom element by id
    const getById = queryByAttribute.bind(null, 'id');
    const form = getById(container, 'all');
    //mock custom event called inside onSubmit handler
    customEvent.emit = jest.fn();
    fireEvent.submit(form);
    /**
     * react testing librarary when creates component
     * it doesnt track any methods attached as  <form onSubmit={handleSubmitAll} >
     * thefore even its being called you can test against it as
     * expect(handleSubmitAll).toHaveBeenCalledTimes(1)
     * but you can test whatever is called inside thefore the below
     */

    expect(customEvent.emit).toHaveBeenCalledTimes(1);
  });
});
