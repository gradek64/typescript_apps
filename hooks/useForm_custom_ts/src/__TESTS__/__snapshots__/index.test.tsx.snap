// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correclty with initial state should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="helloworld"
  >
    <h2>
      AddressForm validation
    </h2>
    <form
      id="addressForm"
    >
      <label
        for="surname"
      >
        surname
      </label>
      <input
        class=""
        data-error=""
        name="surname"
        type="text"
        value=""
      />
      <label
        for="name"
      >
        name
      </label>
      <input
        class=""
        data-error=""
        name="name"
        type="text"
        value=""
      />
      <label
        for="about"
      >
        ABOUT
      </label>
      <textarea
        class=""
        data-error=""
        name="about"
        placeholder="describe yourself..."
      />
      <div
        class="radioButtons "
      >
        <div>
          <label>
            0-30
          </label>
          <input
            data-error=""
            name="age"
            type="radio"
            value="30"
          />
        </div>
        <div>
          <label>
            31-60
          </label>
          <input
            data-error=""
            name="age"
            type="radio"
            value="60"
          />
        </div>
        <div>
          <label>
            61-90
          </label>
          <input
            checked=""
            data-error=""
            name="age"
            type="radio"
            value="90"
          />
        </div>
      </div>
      <label
        for="telephone"
      >
        telephone
      </label>
      <input
        class=""
        data-error=""
        name="telephone"
        type="number"
        value=""
      />
      <label
        for="hobby"
      >
        hobby
      </label>
      <input
        class=""
        data-error=""
        name="hobby"
        type="text"
        value=""
      />
      <input
        type="submit"
        value="submit"
      />
    </form>
    <h2>
      BillingForm validation
    </h2>
    <form
      id="accountForm"
    >
      <label
        for="surname"
      >
        surname
      </label>
      <input
        class=""
        data-error=""
        name="surname"
        type="text"
        value=""
      />
      <label
        for="name"
      >
        name
      </label>
      <input
        class=""
        data-error=""
        name="name"
        type="text"
        value=""
      />
      <label
        for="paymentMethod"
      >
        payment Method
      </label>
      <select
        class=""
        data-error=""
        name="paymentMethod"
      >
        <option
          disabled=""
          selected=""
          style="color: gray;"
          value=""
        >
          pick a payment method
        </option>
        <option
          value="visa card"
        >
          visa card
        </option>
        <option
          value="mastercard"
        >
          mastercard
        </option>
        <option
          value="paypal"
        >
          paypal
        </option>
        <option
          value="klarna"
        >
          klarna
        </option>
      </select>
      <label
        for="accountNumber"
      >
        accountNumber
      </label>
      <input
        class=""
        data-error=""
        name="accountNumber"
        type="number"
        value=""
      />
      <input
        type="submit"
        value="submit"
      />
    </form>
    <div
      id="submitAll"
    >
      <form
        id="all"
      >
        <input
          type="submit"
          value="submit"
        />
      </form>
    </div>
  </div>
</DocumentFragment>
`;
